src/main/java/juja/microservices/teams/service/TeamService.java

```diff
@@ -3,54 +3,84 @@
 import juja.microservices.teams.dao.TeamRepository;
 import juja.microservices.teams.entity.Team;
 import juja.microservices.teams.entity.TeamRequest;
-import juja.microservices.teams.exceptions.TeamUserExistsException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import juja.microservices.teams.exceptions.UserExistsException;
+import lombok.extern.slf4j.Slf4j;
+import juja.microservices.teams.exceptions.UserInSeveralTeamsException;
+import juja.microservices.teams.exceptions.UserNotInTeamException;
 import org.springframework.stereotype.Service;
 
 import javax.inject.Inject;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 /**
  * @author Andrii.Sidun
+ * @author Ivan Shapovalov
  */
-
 @Service
+@Slf4j
 public class TeamService {
-
-    private final Logger logger = LoggerFactory.getLogger(this.getClass());
-
     @Inject
     private TeamRepository teamRepository;
 
-    public String addTeam(TeamRequest teamRequest){
-        logger.debug("Start TeamService.addTeam. Team: {}", teamRequest);
-        if (isAnyBodyInOtherTeam(teamRequest)){
-            logger.warn("Some user(s) exists in a team");
-            throw new TeamUserExistsException("Some user(s) exists in a team");
+    public Team addTeam(TeamRequest teamRequest) {
+        log.debug("Started 'addTeam' TeamRequest: {}", teamRequest);
+        Set<String> usersInTeams = usersInCurrentTeams(teamRequest);
```

Comment:
I think it could be simplier - the loop in this method and invoke repository method, check size of result List<Teams>, if more than 0 throw Exception.
without HashSet.